docs/reference/elasticsearch/rest-apis/update-document.md

```diff
@@ -0,0 +1,264 @@
+}
+```
+% TEST[continued]
+
+The following script removes a subfield from an object field:
+
+```console
+PUT test/_doc/1?refresh
+{
+  "my-object": {
+    "my-subfield": true
+  }
+}
+```
+
+```console
+POST test/_update/1
+{
+  "script": "ctx._source['my-object'].remove('my-subfield')"
+}
+```
+% TEST[continued]
+
+Instead of updating the document, you can also change the operation that is
+executed from within the script. For example, this request deletes the doc if
+the `tags` field contains `green`, otherwise it does nothing (`noop`):
+
+```console
+POST test/_update/1
+{
+  "script": {
+    "source": "if (ctx._source.tags.contains(params.tag)) { ctx.op = 'delete' } else { ctx.op = 'noop' }",
+    "lang": "painless",
+    "params": {
+      "tag": "green"
+    }
+  }
+}
+```
+
+## Update part of a document [update-part-document]
+
+The following partial update adds a new field to the
+existing document:
+
+```console
+POST test/_update/1
+{
+  "doc": {
+    "name": "new_name"
+  }
+}
+```
+
+If both `doc` and `script` are specified, then `doc` is ignored. If you
+specify a scripted update, include the fields you want to update in the script.
+
+
+## Detect noop updates [detect-noop-updates]
+
+By default updates that don't change anything detect that they don't change
+anything and return `"result": "noop"`:
+
+```console
+POST test/_update/1
+{
+  "doc": {
+    "name": "new_name"
+  }
+}
+```
+% TEST[continued]
+
+If the value of `name` is already `new_name`, the update
+request is ignored and the `result` element in the response returns `noop`:
+
+```console
+
+{
+   "_shards": {
+        "total": 0,
+        "successful": 0,
+        "failed": 0
+   },
+   "_index": "test",
+   "_id": "1",
+   "_version": 2,
+   "_primary_term": 1,
+   "_seq_no": 1,
+   "result": "noop"
+}
+```
+
+You can disable this behavior by setting `"detect_noop": false`:
+
+```console
+POST test/_update/1
+{
+  "doc": {
+    "name": "new_name"
+  },
+  "detect_noop": false
+}
+```
+
+## Upsert [upsert]
+
+An upsert operation lets you update an existing document or insert a new one if it doesn't exist, in a single request.
+
+In this example, if the product with ID `1` exists, its price will be updated to `100`. If the product does not exist, a new document with ID `1` and a price of `50` will be inserted.
+
+```console
+POST /test/_update/1
+{
+  "doc": {
+    "product_price": 100
+  },
+  "upsert": {
+    "product_price": 50
+  }
+}
+```
+
+## Scripted upsert [scripted-upsert]
+
+To run the script whether or not the document exists, set `scripted_upsert` to
+`true`:
+
+```console
+POST test/_update/1
+{
+  "scripted_upsert": true,
+  "script": {
+    "source": """
+      if ( ctx.op == 'create' ) {
+        ctx._source.counter = params.count
+      } else {
+        ctx._source.counter += params.count
+      }
+    """,
+    "params": {
+      "count": 4
+    }
+  },
+  "upsert": {}
+}
+```
+
+## Doc as upsert [scripted-upsert]
```

Comment:
Here you've got a doc id duplication `[scripted-upsert]`

Reply:
Can improve heading here to explain what user value is and use verb phrase for parallel structure